### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by tonimueller**
I changed the structure of the CMakeLists.txt  to top-down which is more logical for me. The top CMakeLists.txt takes now an option for the test (-DLEMONADE_TEST=ON/OFF) and the standard option for the lemonade install path of the main repo (-DLEMONADE_DIR=path/to/install/ ).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,31 +12,31 @@
...
+  ./testsLeMonADE
+  `````
+* Please write tests although it might appear annoying but it helps you to find nasty bugs and tests the correctness of your algorithms!
+
+## Contributions
+
+* Any contributions making the programm better are welcome as well as hints for bugs.
+* Every contribution should be made by following the guidelines made by the [gitflow workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow)
 
-* testing might be annoying but helps you to find nasty bugs
```
Comment (MartinWenge) [inline README.md:40]: Thanks for the reformulation of my harsh statement :cactus:

File: projects/exampleProject/CMakeLists.txt [hunk 2]
```diff
@@ -1,25 +1,5 @@
 cmake_minimum_required (VERSION 2.8)
```
Comment (MartinWenge) [inline projects/exampleProject/CMakeLists.txt]: This line can be omitted?

File: projects/examplesLinearChains/CMakeLists.txt [hunk 3]
```diff
@@ -1,33 +1,5 @@
 cmake_minimum_required (VERSION 2.8)
```
Comment (MartinWenge) [inline projects/examplesLinearChains/CMakeLists.txt]: Omit this line as well? Minimum version is defined in the top level CMakeList

File: test/CMakeLists.txt [hunk 4]
```diff
@@ -1,22 +1,5 @@
 cmake_minimum_required (VERSION 2.8)
```
Comment (MartinWenge) [inline test/CMakeLists.txt]: Omit this as well.

**Review by MartinWenge — APPROVED** [review]
Could you omit the minium required lines in the low level CMakeLists? Then it is perfect :smiley_cat:

**Comment by tonimueller** [comment]
Thanks for the changes!

**Pull request merged**